### PR TITLE
gcworker: fix potential gcworker goroutine leak during tikv down

### PR DESCRIPTION
**Pull request by lysu**
### What problem does this PR solve? 

current gcworker renew backoff for each retry, this maybe lead to infine retry when kv down. 

### What is changed and how it works?

renew backoff when success and break loop when backoff many times.

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

 - impl change

Side effects

 - goroutine will keep retry if kv isn't up

Related changes

 - Need to cherry-pick to the release branch 3.0

Release note

 - fix potential gcworker goroutine leak during tikv down

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/13921)

### CODE REVIEW

**Comment by tiancaiamao** [comment]
LGTM

**Comment by jackysp** [comment]
PTAL @MyonKeminta

File: store/tikv/gcworker/gc_worker.go [hunk 1]
```diff
@@ -888,6 +894,11 @@ func (w *GCWorker) resolveLocksForRange(ctx context.Context, safePoint uint64, s
 			continue
 		}
 
+		bo = tikv.NewBackoffer(ctx, tikv.GcResolveLockMaxBackoff)
+		failpoint.Inject("setGcResolveMaxBackoff", func(v failpoint.Value) {
+			sleep := v.(int)
+			bo = tikv.NewBackoffer(ctx, sleep)
+		})
```
Comment (MyonKeminta) [inline store/tikv/gcworker/gc_worker.go]: Can it be moved to below line 916? So the last loop will go to break before creating a useless backoffer.

**Pull request merged**